Tests/SentryTests/Protocol/SentryLogTests.swift

```diff
@@ -0,0 +1,107 @@
+@testable import Sentry
+import XCTest
+
+final class SentryLogTests: XCTestCase {
+    
+    // MARK: - Encoding Tests
+    
+    private let log = SentryLog(
+        timestamp: Date(timeIntervalSince1970: 1_234_567_890),
+        traceId: SentryId(uuidString: "550e8400-e29b-41d4-a716-446655440000"),
+        level: SentryLog.Level.info,
+        body: "Test log message",
+        attributes: [
+            "user_id": .string("12345"),
+            "is_active": .boolean(true),
+            "count": .integer(42),
+            "score": .double(3.14159)
+        ],
+        severityNumber: 21
+    )
+    
+    private let jsonData = Data("""
+    {
+        "timestamp": "2009-02-13T23:31:30.000Z",
+        "trace_id": "550e8400e29b41d4a716446655440000",
+        "level": "trace",
+        "body": "Trace message",
+        "attributes": {
+            "key1": {
+                "type": "string",
+                "value": "value1"
+            },
+            "key2": {
+                "type": "boolean",
+                "value": false
+            }
+        },
+        "severity_number": 21
+    }
+    """.utf8)
+    
+    func testEncode() throws {
+        let data = try encodeToJSONData(data: log)
+        let json = try JSONSerialization.jsonObject(with: data) as? [String: Any]
+        
+        XCTAssertNotNil(json)
```

Comment:
`l`: You can again use XCTUnwrap to avoid having to call `XCTAssertNotNil` and then you can ditch all the ? below, so `json?["trace_id"]` becomes `json["trace_id"]`.